wob.c

```diff
@@ -292,6 +294,13 @@ main(int argc, char **argv)
 					return EXIT_FAILURE;
 				}
 				break;
+			case 'm':
+				maximum = (uint16_t) atoi(optarg);
+				if (maximum < 1 || maximum > UINT16_MAX){
```

Comment:
I don't think that `maximum` can be larger than `UINT16_MAX` as it is `uint16_t` type. I think you could set maximum value for `-m` to something smaller than `UINT16_MAX` and check if it is larger than that value.

Something like:
```suggestion
				if (maximum < 1 || maximum > 1000) {
```

Reply:
How do you feel about doing something like this? :
```
case 'm':
	if (atoi(optarg) < 1 || atoi(optarg) > UINT16_MAX){
	fprintf(stdout, "Maximum must be a value between 1 and %d \n", UINT16_MAX);
		return EXIT_FAILURE;
	}
	maximum = (uint16_t) atoi(optarg);
	break;
```
This solves the issue of negative numbers and warns the user if he attempts to input a larger value than UINT16_MAX. i.e. we compare the number before casting.

Reply:
Too complicated for my taste. Leave the maximum unchecked for now and we will get back to it later.

Reply:
@alright